agent/src/main/java/com/newrelic/agent/android/sessionReplay/SessionReplay.java

```diff
@@ -133,67 +151,65 @@ public static void deInitialize() {
 
     @Override
     public void onHarvestStart() {
-        // No-op
+        // Only prepare for harvest when in FULL mode
+        if (modeManager != null && modeManager.getCurrentMode() == SessionReplayMode.FULL) {
+            // Mark that harvest is starting - pause frame writes to prevent race condition
+            isHarvesting.set(true);
+            log.debug("SessionReplay: Harvest started, pausing frame writes");
+        }
     }
 
     @Override
     public void onHarvest() {
-        log.debug("Harvest started, processing frames and touch data");
-        if (rawFrames.isEmpty() && touchTrackers.isEmpty()) {
-            Log.d("SessionReplay", "No frames or touch data to process.");
+        // Only harvest when in FULL mode
```

Comment:
Does this need to check for OFF as well?

Reply:
no, if user pause the replay and we have some frames already on the file,those frames should be harvested .